server.js

```diff
@@ -1,25 +1,38 @@
+import cluster from 'cluster';
+import os from 'os';
 import fastify from "fastify";
 import drawCanvas from "./draw-canvas.js";
 import fetch from 'node-fetch';
 
+const numCPUs = os.cpus().length;
 const server = fastify({ logger: true });
 
-server.get("/*", async (request, reply) => {
-  if (request.params["*"] === "favicon.ico") return reply.send("");
-
-  let backgroundImage = null;
-  if (request.query.bg) {
-    const res = await fetch(request.query.bg);
-    const buffer = await res.buffer();
-    backgroundImage = `data:${res.headers['content-type']};base64,${buffer.toString('base64')}`;
+if (cluster.isMaster) {
+  console.log(`Master ${process.pid} is running`);
+  for (let i = 0; i < numCPUs; i++) {
+    cluster.fork();
   }
+  cluster.on('exit', worker => {
+    console.log(`Worker ${worker.process.pid} died`);
```

Comment:
Wondering if this needs to refork the worker so that we don’t end up with no workers running

Reply:
also what happens if master dies?